example-app/src/main/java/com/adyen/checkout/example/ui/settings/composable/SettingsScreen.kt

```diff
@@ -0,0 +1,267 @@
+import com.adyen.checkout.example.ui.settings.model.SettingsCategory
+import com.adyen.checkout.example.ui.settings.model.SettingsIdentifier
+import com.adyen.checkout.example.ui.settings.model.SettingsItem
+import com.adyen.checkout.example.ui.settings.model.SettingsUIState
+import com.adyen.checkout.example.ui.settings.viewmodel.SettingsViewModel
+import com.adyen.checkout.example.ui.theme.ExampleTheme
+
+@Composable
+internal fun SettingsScreen(
+    onBackPressed: () -> Unit,
+    viewModel: SettingsViewModel = hiltViewModel(),
+) {
+    Scaffold(
+        modifier = Modifier.windowInsetsPadding(WindowInsets.ime),
+        topBar = {
+            TopAppBar(
+                title = { Text(text = stringResource(id = R.string.settings_screen_title)) },
+                navigationIcon = {
+                    IconButton(onClick = onBackPressed) {
+                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Back")
+                    }
+                },
+            )
+        },
+    ) { innerPadding ->
+        val uiState by viewModel.uiState.collectAsState()
+        SettingsScreen(
+            uiState = uiState,
+            onItemClicked = viewModel::onItemClicked,
+            onEditSettingConsumed = viewModel::onEditSettingConsumed,
+            onTextSettingChanged = viewModel::onTextSettingChanged,
+            onListSettingChanged = viewModel::onListSettingChanged,
+            onSwitchSettingChanged = viewModel::onSwitchSettingChanged,
+            modifier = Modifier.padding(innerPadding),
+        )
+    }
+}
+
+@Suppress("LongParameterList")
+@Composable
+private fun SettingsScreen(
+    uiState: SettingsUIState,
+    onItemClicked: (SettingsItem) -> Unit,
+    onEditSettingConsumed: () -> Unit,
+    onTextSettingChanged: (SettingsIdentifier, String) -> Unit,
+    onListSettingChanged: (SettingsIdentifier, EditSettingsData.SingleSelectList.Item) -> Unit,
+    onSwitchSettingChanged: (SettingsIdentifier, Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    SettingsItemsList(
+        settingsCategories = uiState.settingsCategories,
+        onItemClicked = onItemClicked,
+        onSwitchSettingChanged = onSwitchSettingChanged,
+        modifier = modifier,
+    )
+
+    if (uiState.settingToEdit != null) {
+        EditSettingDialog(
+            settingToEdit = uiState.settingToEdit,
+            onTextSettingChanged = {
+                onTextSettingChanged(uiState.settingToEdit.identifier, it)
+                onEditSettingConsumed()
+            },
+            onListSettingChanged = {
+                onListSettingChanged(uiState.settingToEdit.identifier, it)
+                onEditSettingConsumed()
+            },
+            onDismiss = {
+                onEditSettingConsumed()
+            },
+        )
+    }
+}
+
+@Composable
+private fun SettingsItemsList(
+    settingsCategories: List<SettingsCategory>,
+    onItemClicked: (SettingsItem) -> Unit,
```

Comment:
`onItemClicked` suggests that in every item click this callback will be fired, but in the code it only happens for Text setting items. What do you think of changing the name to something like `onTextSettingClicked`?